packages/app/src/components/AddressInput.tsx

```diff
@@ -0,0 +1,107 @@
+'use client'
+import { useState } from 'react'
+import { isAddress } from 'viem'
+import { normalize } from 'viem/ens'
+import { mainnet, sepolia } from 'wagmi/chains'
+import { createConfig, useEnsAddress, http, useEnsAvatar } from 'wagmi'
+import Image from 'next/image'
+
+interface AddressInputProps extends React.HTMLProps<HTMLInputElement> {
+  onRecipientChange: (address: string, isValid: boolean) => void
+  onRawInputChange?: (address: string) => void
+}
+// wagmi config
+export const config = createConfig({
+  chains: [mainnet, sepolia],
+  transports: {
+    [mainnet.id]: http(),
+    [sepolia.id]: http(),
+  },
+})
+
+function truncateAddress(address: string) {
+  return `${address.slice(0, 9)}...${address.slice(-9)}`
+}
+
+export const AddressInput = ({ onRecipientChange, onRawInputChange }: AddressInputProps) => {
+  const [isValidToAddress, setIsValidToAddress] = useState<boolean>(false)
+  const [rawTokenAddress, setRawTokenAddress] = useState<string>('')
+  let name
+
+  try {
+    name = normalize(rawTokenAddress)
```

Comment:
It could be interesting to move most of this to a helper/utils function as well.

Input: address
Function: normalize address, fetch name, fetch avatar, maybe even other profile info/data?
Output: ENS profile info 

Happy to accept for this PR, but worth thinking about if you want to continue :)

Reply:
I can def keep working on this. Would rather make it good than just have you accept it lol

Will re-request review when i get a chance to make the changes

Reply:
Hah, appreciate that. Multiple, small iterative PRs can be fine as well :)

But sounds good!